src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsPresenter.cs

```diff
@@ -324,9 +319,14 @@ internal void LoadChildren(_ViewGroup panel)
 					HorizontalContentAlignment = HorizontalAlignment.Stretch,
 					IsTabStop = false
 				};
-
-				VisualTreeHelper.AddChild(this, FooterContentControl);
 			}
+
+			// We want FooterContentControl to be assigned before calling SetItemsPanel,
+			// because it may directly cause an ArrangeOverride to be called in some cases,
+			// where the value is expected to be non-null.
+			if (HeaderContentControl is { }) { VisualTreeHelper.AddChild(this, HeaderContentControl); }
+			SetItemsPanel(panel);
+			if (FooterContentControl is { }) { VisualTreeHelper.AddChild(this, FooterContentControl); }
```

Comment:
@Xiaoy312 if we can't make a runtime test for this, can we make a manual test?

Reply:
I wasnt able to reproduce this neither. In theory, just a simple GridView with ItemTemplate and ItemsSource should be enough to repro, which we have plenty of examples under the "GridView" sample category. Yet, it just work correctly in uno.